### ISSUE TITLE
New agent: cannot set random password, cannot reset password. Manual password only

### ISSUE THREAD

**Issue by LeCoyote** [osTicket/osTicket#4860]
### Prerequisites

* [ ] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [ ] Do you have any errors in the PHP error log, or javascript console?
* [x] Did you check the [osTicket forums](http://osticket.com/forum/categories)?
* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

For more information on how to write a good [bug report](https://github.com/osTicket/osTicket/wiki/Github-Issues-Guidelines)

### Description

(This looks like a bug to me, but it might also be that the documentation is lacking and/or the UI is misleading.)

When creating a new agent, as a default, osTicket offers to send a "reset password" email to the agent. However, it does no such thing as the agent is actually created (ie. no email is sent to them). Furthermore, the newly created agent cannot use the "forgot password" link: osTicket will simply claim that it cannot and to contact an administrator. The administrator's "reset password" facility, when used on the new agent, briefly opens and closes a window before going back to the password setup window, and still nothing is send via email. There is no error log anywhere.

The only thing that works is to manually set a password for the agent. If this is by design, then the default behaviour should be to ask for a manual password. If not, then this is a bug that should be fixed.

As soon as the agent logs in using the **manually set** password, then all the "reset password" options start working as expected.

### Steps to Reproduce

1. Create an agent with every setting left on default regarding the password

**Expected behavior:** osTicket should generate a password and send it via email

**Actual behavior:** no message is sent, and the administrator is not informed either.

### Versions

osTicket v1.11 (d4e240b)
PHP 7.2.17
nginX 1.14.2
MariaDB 10.3.14

All required PHP extensions are installed and enabled. Outbound email is configured, tested, and works as expected.

**Comment by JediKev** [comment]
@LeCoyote

So I’ve only seen this a few times and never got appropriate details from the reporting party to look into it fully.

What are the errors in you PHP/Apache error logs?

Cheers.

**Comment by LeCoyote** [comment]
@JediKev That's the problem: absolutely no errors are logged anywhere, not even in osTicket's own debug log.
I'm not sure that nginX is a factor here. The fact that the password reset fails at first, then starts working as soon as the user has logged in once, would tend to confirm that the problem is elsewhere, IMHO.
I haven't investigated where, in the code, is the password generation/sending mechanism, perhaps we could start there and add some debugging stuff around to see what happens?
Thanks.

**Comment by JediKev** [comment]
@LeCoyote

Not necessarily...you’d be surprised at the amount of issues that appeared to be code related but were actually due to NGINX.

I’ll look into this when I get back to work. Hopefully I’ll be able replicate this with NGINX.

Cheers.

**Comment by LeCoyote** [comment]
@JediKev 
Let's call it a gut feeling, but I'm ready to be proven wrong :) Thanks for looking into it.

**Comment by JediKev** [comment]
@LeCoyote

Lol I’m not doubting you just saying don’t ever rule anything out as it’s almost always the least expected things that wind up being the culprit in the end :)

Cheers.

**Comment by matzneda** [comment]
I can confirm the exact same behaviour happens on Apache

osTicket Version v1.11 (d4e240b)
Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/7.2.16
MySQL Version 5.7.26
PHP Version 7.2.16

**Comment by osticketeer** [comment]
I can confirm a very simmular behaviour happens on Ubuntu LAMP Server.

First some info:
it's a new installation on a new ubuntu vm using Ubuntu 18.04.3 LTS

PHP 7.2.19-0ubuntu0.18.04.2 (cli) (built: Aug 12 2019 19:34:28) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.19-0ubuntu0.18.04.2, Copyright (c) 1999-2018, by Zend Technologies

Apache/2.4.29 (Ubuntu)

mysql  Ver 14.14 Distrib 5.7.27, for Linux (x86_64) using  EditLine wrapper

I'm using the Git version and replicating once a day with:
git pull
php manage.php deploy -v /var/www/html/support/


PHP Extensions
--
gdlib | Used for image manipulation and PDF printing
imap | Used for email fetching
xml | XML API
xml-dom | Used for HTML email processing
json | Improves performance creating and processing JSON
mbstring | Highly recommended for non western european language content
phar | Highly recommended for plugins and language packs
intl | Highly recommended for non western european language content
fileinfo | Used to detect file types for uploads
APCu | Improves overall performance
Zend Opcache | Improves overall performance

cgi.fix_pathinfo | "1" is recommended if AJAX is not working
date.timezone | Europe/Berlin

Schema | osticker2019 (localhost)
Schema Signature | 00c949a623b82848baaf3480b51307e3
Space Used | 2.80 MiB
Space for Attachments | 0.09 MiB
Timezone | UTC

^The PHP-Modules are installed and checked.

Installed Language Packs
Deutsch (German) — de_DE  
de — include/i18n/de.phar
Version: 154948, for version v1.10
Built: Wed, 06 Feb 19 21:18:23 +0000

English (United States)
en_US — include/i18n/en_US

For debuging the German Language Pack is deactivated...



I''m unable to set a Agent password manually. Clicking on the Set-Password button results in nothing.
The Agent is unable to reset the Password using the Reset-Password feature at login. 
This is reproduceable with every new Agentaccount. The only Agenaccount that works is mine, the first one.

I would like to debug osTicket and have set the "Default Log Level" to "Debug" but that's just not what i need. Well i need some help.

Please tell me wich additional informations are needed.

Thanks

**Comment by ntozier** [comment]
1.11 support was discontinued.  Please upgrade to current (1.12.2 as of this response) and see if you still have the problem.

**Comment by osticketeer** [comment]
Thanks for answering, i'm using the git-Version: 1.12-git (2313dd8)

**Comment by osticketeer** [comment]
I've found out the the not working Agents had a NULL - Value in <osticket-db>\ost_staff\passwd column. So i set it to the value of the first created and working Agent and that did the trick. The not working Agents can now use the reset-password-function.

For other folks who have that problem: 
1. mysql -u root
2. show databases;
3. use <your osticket-db>;
4. select * from ost_staff;

To set a password:
UPDATE ost_staff SET passwd = "$2a$078$y7o4DCObsIU5YfdO6pJveu6FGtqweqAjznM8f/dPepW9gkSvEIoNtS" WHERE lastname = "<Agents-Lastname>";

I hope that helps...

This is a workaround and the question still exists: Why does this happen ?

**Comment by osticketeer** [comment]
This Ticketsystem eats up things that are in airrow-signs...

3. use your_osticket_db;

UPDATE ost_staff SET passwd = "$2a$078$y7o4DCObsIU5YfdO6pJveu6FGtqweqAjznM8f/dPepW9gkSvEIoNtS" WHERE lastname = "not_working_agents_lastname";

**Comment by jakobeissler** [comment]
I have the same problem with version v1.14.1. Workaround suggested by @osticketeer worked.

**Comment by JediKev** [comment]
@jakobeissler 

Have you tried these patches yet?
- https://github.com/osTicket/osTicket/pull/4968
- https://github.com/osTicket/osTicket/pull/5308

Cheers.

**Comment by jakobeissler** [comment]
@JediKev nope, i'm using docker image from: https://hub.docker.com/r/campbellsoftwaresolutions/osticket

**Comment by JediKev** [comment]
@jakobeissler 

Apply the patch and you should be good.

Cheers.

**Comment by jakobeissler** [comment]
@JediKev I will do so. Thank you very much.
